### DISCUSSION TITLE
Serialization of pydantic models on publish

### DISCUSSION THREAD

**Discussion by bodograumann** [Lancetnik/Propan#74] [Ideas]
I am trying to send a pydantic object with broker.publish and it is unable to serialize datetime fields:

> TypeError: Object of type datetime is not JSON serializable

I think this part of the code is relevant:
https://github.com/Lancetnik/Propan/blob/main/propan/brokers/_model/schemas.py#L199

When I encode it myself beforehand, datetime fields are handled automatically by pydantic, e.g. `broker.publish(pydantic_object.json(), content_type="application/json", …)`.

Can you also use the pydantic json-serialization logic automatically here?

PS: A similar issue could exist when returning a pydantic object from a handler method on the server side.

**Comment by Lancetnik** [comment]
It's a funny, but it is the most suitable way to use `pydantic.json` to serialize response

```python
if isinstance(m, (Dict, Sequence)):
    return obj.json()[12:-1].encode(), ContentTypes.json.value
```

**Comment by Lancetnik** [comment]
So, I got an elegant solution to serialize publishing messages. With handlers should be any problems cuz they are serializing at **FastDepends** by pydantic model.

**Reply by bodograumann** [reply]
Yes, just switched to the fastapi integration and both serialization and deserialization work prefectly :heart:

**Reply by Lancetnik** [reply]
@bodograumann  Are you using FastAPI Router to serialize incoming messages? But I improved serialization with the last update... Now It uses pydantic json to publish messages and tested at all of [these cases](https://github.com/Lancetnik/Propan/blob/main/tests/brokers/base/publish.py#L21).

**Reply by bodograumann** [reply]
I mean I am now using the propan fastapi plugin and just wanted to say it works very nicely. Thanks for your great work!
